osquery/events/linux/auditdnetlink.cpp

```diff
@@ -857,32 +877,67 @@ bool AuditdNetlinkParser::ParseAuditReply(
 
   // Parse the record header
   event_record.type = reply.type;
-  boost::string_ref message_view(reply.message,
-                                 static_cast<unsigned int>(reply.len));
+  std::string_view message_view(reply.message,
+                                static_cast<std::size_t>(reply.len));
+
+  /* NOTE: The preamble is "audit(%llu.%03lu:%u): ",
+     this means that there are 6 fixed characters,
+     1 to 19 characters for the time, 3 characters for the nanoseconds,
+     1 to 10 characters for the serial, 2 for the separators. */
 
-  auto preamble_end = message_view.find("): ");
-  if (preamble_end == std::string::npos) {
+  // Verify that the message starts with the correct prefix
+  if (message_view.compare(
+          0, kAuditPreamblePrefix.size(), kAuditPreamblePrefix) != 0) {
+    return false;
+  }
+
+  /* Verify that the preamble end is found, that there are enough characters
+     for the smallest preamble and that the preamble isn't
+     bigger than the biggest one possible. */
+  auto preamble_end = message_view.find("): ", kAuditPreamblePrefix.size());
+  if (preamble_end < kAuditMinPreambleEndPos ||
+      preamble_end > kAuditMaxPreambleEndPos) {
+    return false;
+  }
+
+  // The time should end at least 1 character beyond the end of the prefix
+  auto time_end = message_view.find(".", kAuditPreamblePrefix.size() + 1);
+
+  /* Make sure that the time doesn't somehow consume all the characters.
+     A minimum of 5 characters should be present before the end of the
+     preamble, for the nanoseconds,
+     the separator and the shortest possible serial. */
+  if (time_end >= (preamble_end - 5)) {
     return false;
   }
 
   event_record.time =
-      tryTo<unsigned long int>(message_view.substr(6, 10).to_string(), 10)
+      tryTo<std::uint64_t>(std::string(message_view.substr(
+                               kAuditPreamblePrefix.size(),
+                               time_end - kAuditPreamblePrefix.size())),
+                           10)
           .takeOr(event_record.time);
-  event_record.audit_id = message_view.substr(6, preamble_end - 6).to_string();
+
+  event_record.audit_id = message_view.substr(
+      kAuditPreamblePrefix.size(), preamble_end - kAuditPreamblePrefix.size());
 
   // SELinux doesn't output valid audit records; just save them as they are
-  if (IsSELinuxRecord(reply)) {
-    event_record.raw_data = reply.message;
+  if (subtype == AuditRecordSubtype::SELinux) {
+    event_record.raw_data = message_view;
     return true;
   }
 
   // Save the whole message for AppArmor too
-  if (isAppArmorRecord(reply)) {
-    event_record.raw_data = reply.message;
+  if (subtype == AuditRecordSubtype::AppArmor) {
+    event_record.raw_data = message_view;
+  }
+
+  if (message_view.size() < (preamble_end + 3)) {
+    return false;
```

Comment:
With the addition of this `return false` and the stricter check that can result in the `return false` on line 911, is there an error logged? (In case this code ever started discarding too many Audit messages.)